### ISSUE TITLE
Fabel testing: number field

### ISSUE THREAD

**Issue by connorscarolyns** [cds-snc/platform-forms-client#4851]
## Description
entry is being read oddly.

Voiveover in Safari

https://app.makeitfable.com/share/675ae655e9c0df46797459d2

**Comment by thiessenp-cds** [comment]
I was unable to reproduce the same issue the user encountered. I tried entering new numbers and also using the up-down arrow keys to increase-decrease the numbers. I could not get VO to announce the same pattern 0 followed by the new number.

I did find a different bug though in VO. Once the input stepper goes beyond 10, VO will only announce the first digit. For example, 22 would be announced as 2. This could be very confusing for mac users. Oddly it looks like Apple does not have a place to file and follow bug tickets. But they do have an email address - I emailed them...
e.g. https://www.digitala11y.com/how-where-to-report-accessibility-bugs/

Attached are some example videos.

https://github.com/user-attachments/assets/2537e6a7-c12f-44e7-a187-6ae957613848


https://github.com/user-attachments/assets/2a78edca-acf4-4ad9-828b-14634a01eb0b


https://github.com/user-attachments/assets/57b84908-0ea7-4c39-8906-2f61488e309a

**Comment by connorscarolyns** [comment]
I wonder if an improvement might be that we don't use the number stepper field and restrict the input only to accept numbers. There is limited usability benefits to having an incremental stepper, and there seems to be odd behaviour associated with it.

**Comment by thiessenp-cds** [comment]
> I wonder if an improvement might be that we don't use the number stepper field and restrict the input only to accept numbers. There is limited usability benefits to having an incremental stepper, and there seems to be odd behaviour associated with it.

Makes sense to me and also looks like the change would be inline with what the [gov.uk design system](https://technology.blog.gov.uk/2020/02/24/why-the-gov-uk-design-system-team-changed-the-input-type-for-numbers/) is doing. @timarney can you see any problem with this change or additional background we may not know?

The solution could be replacing the 
```
<input  type="number">
with something like 
<input type="text" inputmode="numeric" pattern="[0-9]*"> 
```

**Comment by timarney** [comment]
No concerns with not using a stepper

**Comment by ioanacontu** [comment]
@thiessenp-cds to make suggested changes.
